app/client/components/GristClientSocket.ts

```diff
@@ -152,8 +158,12 @@ export class GristClientSocketWS extends GristClientSocket {
     }
   }
 
-  public set onmessage(cb: null | ((event: { data: string }) => void)) {
-    this._ws.onmessage = cb;
+  public set onmessage(cb: null | ((data: string) => void)) {
+    this._ws.onmessage = cb ?
+      (event: WS.MessageEvent | MessageEvent<any>) => {
+        cb(event.data instanceof String ? event.data : event.data.toString());
```

Comment:
On the client side, I think it is supposed to always be a `string` since we are only sending text frames, see https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/message_event#event_properties
> If the message type is "text", then this field is a string.

On the server side (using `ws`), the call to `.toString()` here is meant to only make explicit what `JSON.parse` in `GristWSConnection` was doing implicitly.

But in fact, [looking at what `ws` does to emulate the HTML5 WebSocket API](https://github.com/websockets/ws/blob/5e42cfdc5fa114659908eaad4d9ead7d5051d740/lib/event-target.js#L203-L210), it seems the `toString()` is not even necessary since it will have been node by `ws` to convert its NodeJS `Buffer` object (which has a useful `toString`, as contrasted from `ArrayBuffer` or `Blob`) to a string to match standard WebSocket behavior.

So I've gone ahead and removed the `toString()` and added an explanatory comment.

```suggestion
        // event.data is guaranteed to be a string here because we only send text frames.
        // https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/message_event#event_properties
        cb(event.data);
```